src/core/algorithms/statistics/data_stats.cpp

```diff
@@ -606,6 +623,295 @@ Statistic DataStats::GetAvgNumberOfChars(size_t index) const {
     return Statistic(res, &double_type, false);
 }
 
+Statistic DataStats::GetMinChars(size_t index) const {
+    if (all_stats_[index].min_chars.HasValue()) return all_stats_[index].min_chars;
+    mo::TypedColumnData const &col = col_data_[index];
+    if (col.GetTypeId() != +mo::TypeId::kString) return {};
+
+    mo::IntType int_type;
+    std::size_t result = std::numeric_limits<std::size_t>::max();
+    for (size_t i = 0; i < col.GetNumRows(); i++) {
+        if (col.IsNullOrEmpty(i)) continue;
+
+        auto const &string_data = mo::Type::GetValue<std::string>(col.GetValue(i));
+        std::size_t const &size = string_data.size();
+
+        if (size < result) result = size;
+    }
+
+    std::byte const *res = int_type.MakeValue(result);
+
+    return Statistic(res, &int_type, false);
```

Comment:
Generalize this code in a private method:
```
return GetMinStringStatistic(index, [](const std::string &line) {
    return line.size();
});
```
Use it for `GetMinNumerOfChars`, `GetMinNumerOfWords`.
Same for Max statistics.